tokio/src/sync/semaphore.rs

```diff
@@ -620,6 +620,19 @@ impl<'a> SemaphorePermit<'a> {
     pub fn forget(mut self) {
         self.permits = 0;
     }
+
+    /// Merge two [`SemaphorePermit`] instances together, consuming `other`
+    /// without releasing the permits it holds.
+    ///
+    /// Permits held by both `self` and `other` are released when `self` drops.
```

Comment:
The docs should mention the conditions when `merge` will panic.

Reply:
Additionally, the method should be marked `#[track_caller]`.

Reply:
Done in 5900f845a418f52c12837fec8acd29923ce827bb - thanks! :+1: